File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>

int main(void) {
    float avg = 0;
    int totalWords = 1;
    int totalSentences = 0;
    string text = get_string("Text: ");
    int i = 0;
    while (text[i] != '\0')
    {
        if (text[i] == ' ')
        {
            totalWords++;
        }
        else if(text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            totalSentences++;
        }
        else if (isalpha(text[i]))
        {
            avg++;
        }
        i++;
    }
    float L = (float)avg / totalWords * 100;
    float S = (float) totalSentences / totalWords * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index <= 16)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: readability.py
import cs50

text = cs50.get_string("Text: ")
letters = 0
words = 1
sentences = 0

for c in text:
    if c.isalpha():
        letters += 1

for c in text:
    if c == " ":
        words += 1

for c in text:
    if c == '.' or c == '!' or c == '?':
        sentences += 1
L = letters * 100 / words
S = sentences * 100 / words
grade = round(0.0588 * L - 0.296 * S - 15.8)
if (grade < 1):
    print("Before Grade 1")
elif (grade <= 16):
    print("Grade", grade)
else:
    print("Grade 16+")
